libs/portals/admin/form-system/src/components/NavbarSelect/components/SelectNavComponent.tsx

```diff
@@ -0,0 +1,148 @@
+import cn from 'classnames'
+import * as styles from './selectNavComponent.css'
+import { Box, Checkbox } from '@island.is/island-ui/core'
+import { useContext } from 'react'
+import {
+  FormSystemGroup,
+  FormSystemInput,
+  FormSystemStep,
+} from '@island.is/api/schema'
+import ControlContext from '../../../context/ControlContext'
+import { ItemType, NavbarSelectStatus } from '../../../lib/utils/interfaces'
+
+type Props = {
+  type: ItemType
+  data: FormSystemStep | FormSystemGroup | FormSystemInput
+  active: boolean
+  selectable: boolean
+}
+
+export default function SelectNavComponent({
+  type,
+  data,
+  active,
+  selectable,
+}: Props) {
+  const { control, selectStatus, controlDispatch, updateSettings } =
+    useContext(ControlContext)
+  const { activeItem, activeListItem, form } = control
+  const activeGuid =
+    selectStatus === NavbarSelectStatus.LIST_ITEM
+      ? activeListItem?.guid ?? ''
+      : activeItem?.data?.guid ?? ''
+  const connected: boolean = form.dependencies[activeGuid]?.includes(
+    data.guid as string,
+  )
+
+  const truncateText = (text: string, maxLength: number) => {
+    return text.length > maxLength ? text.slice(0, maxLength) + '...' : text
+  }
+
+  const truncateName = (name: string) => {
```

Comment:
I notice that this is copy paste around. Please create this a as utility function and reuse it.